src/Traits/SocialLoginTrait.php

```diff
@@ -0,0 +1,110 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Canvas\Traits;
+
+use function Canvas\Core\envValue;
+use function time;
+use Canvas\Models\Sources;
+use Canvas\Models\Users;
+use Canvas\Models\UserLinkedSources;
+use Phalcon\Security\Random;
+
+/**
+ * Trait SocialLoginTrait.
+ *
+ * @package Niden\Traits
+ */
+trait SocialLoginTrait
+{
+    /**
+     * Facebook Login
+     * @param Sources $source
+     * @param string $accessToken
+     * @return Array
+     */
+    protected function facebookLogin(Sources $source, string $accessToken): array
+    {
+
+        /**
+         * Get the Facebook adapter
+         */
+        $facebookAdapter = $this->di->get('socialLogin')->authenticate(ucfirst($source->title));
+
+        /**
+         * Set user's Access Token
+         */
+        $facebookAdapter->setAccessToken($accessToken);
+
+        /**
+         * Get user's profile based on set Access Token
+         */
+        $data = $facebookAdapter->getUserProfile();
+
+        /**
+         * Lets Login or Signup the user
+         */
+        $userProfile = current($data);
+
+        /**
+        * Lets find if user has a linked source by social network id
+        */
+        $userLinkedSource =  UserLinkedSources::findFirst([
```

Comment:
@rwhite27 no need to find the users look for the relationship taht should be good enough